File: client-java/instrumentation/src/test/java/org/evomaster/client/java/instrumentation/example/ExampleUtils.java
package org.evomaster.client.java.instrumentation.example;

import org.evomaster.client.java.instrumentation.shared.ObjectiveNaming;
import org.evomaster.client.java.instrumentation.staticstate.ExecutionTracer;

import java.util.List;
import java.util.Set;
import java.util.function.Consumer;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertTrue;

public class ExampleUtils {

    public static <T> void checkIncreasingTillCoveredForSingleMethodReplacement(
            List<T> inputs,
            T solution,
            Consumer<T> lambda) throws Exception{

        ExecutionTracer.reset();
        assertEquals(0, ExecutionTracer.getNumberOfObjectives());

        double heuristics = -1;
        String target = null;

        for(T val : inputs){
            lambda.accept(val);

            Set<String> missing = ExecutionTracer.getNonCoveredObjectives(ObjectiveNaming.METHOD_REPLACEMENT);
            target = missing.iterator().next();
            assertEquals(1, missing.size());

            double h = ExecutionTracer.getValue(target);
            assertTrue(h >= 0);
            assertTrue(h > heuristics, "Fitness did not improve for: " + val.toString());
            assertTrue(h < 1);
            heuristics = h;
        }

        lambda.accept(solution);

        Set<String> missing = ExecutionTracer.getNonCoveredObjectives(ObjectiveNaming.METHOD_REPLACEMENT);
        assertEquals(0, missing.size());
        double covered = ExecutionTracer.getValue(target);
        assertEquals(1d, covered);
    }
}

File: outdated/outdated-client-js/integration-tests/tests/snippets/methodreplacement/ExampleUtils.js
const ET = require("evomaster-client-js").internal.ExecutionTracer;
const ON = require("evomaster-client-js").internal.ObjectiveNaming;

class ExampleUtils {

    static checkIncreasingTillCoveredForSingleMethodReplacement(inputs, solution, lambda) {

        ET.reset();
        expect(ET.getNumberOfObjectives()).toBe(0);

        let heuristics = -1;
        let target = null;

        for (let k of inputs) {
            lambda(k);

            const missing = ET.getNonCoveredObjectives(ON.METHOD_REPLACEMENT);
            target = Array.from(missing)[0];
            expect(missing.size).toBe(1);

            const h = ET.getValue(target);
            expect(h >= 0).toBe(true);
            expect(h > heuristics).toBe(true);
            expect(h < 1).toBe(true);
            heuristics = h;
        }

        lambda(solution);

        const missing = ET.getNonCoveredObjectives(ON.METHOD_REPLACEMENT);
        expect(missing.size).toBe(0);
        const covered = ET.getValue(target);
        expect(covered).toBe(1);
    }
}

module.exports = ExampleUtils;